rust/src/lockfile.rs

```diff
@@ -18,23 +18,39 @@ use std::io;
 use crate::utils;
 
 /// Parse a JSON lockfile definition.
-fn lockfile_parse_stream<R: io::Read>(input: &mut R,) -> Fallible<LockfileConfig> {
-    let lockfile: LockfileConfig = serde_json::from_reader(input).map_err(|e| {
-        io::Error::new(
-            io::ErrorKind::InvalidInput,
-            format!("serde-json: {}", e.to_string()),
-        )
-    })?;
+fn lockfile_parse_stream<R: io::Read>(
+    fmt: utils::InputFormat,
+    input: &mut R,
+) -> Fallible<LockfileConfig> {
+    let lockfile: LockfileConfig = match fmt {
```

Comment:
Could probably split this out into a generic function that could be shared between `lockfile.rs` and `treefile.rs`, but meh.

Reply:
I tried to but got stuck with generics and lifetimes. I need more oxidizing power before I can pull it off.

Reply:
Ok, I managed to do it now. Added a new commit with the changes.